cmd/createorupdate/update.go

```diff
@@ -43,6 +70,68 @@ func update(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.Plugin)
 	return nil
 }
 
+type scaleOp struct {
+	up   bool
+	down bool
+}
+
+// isScaleOp returns whether the update is a scale up or a scale down
+// for any given role. If there is another unrelated change, the
+// returned map will ne nil, indicating that this operation is not a
+// scaling one.
+func isScaleOp(cs, oldCs *api.OpenShiftManagedCluster) map[api.AgentPoolProfileRole]scaleOp {
+	newCounts := make(map[string]int)
+	for _, new := range cs.Properties.AgentPoolProfiles {
+		newCounts[new.Name] = new.Count
+	}
+
+	scaleOps := make(map[api.AgentPoolProfileRole]scaleOp)
+	var hasScaleOps bool
+	for i, old := range oldCs.Properties.AgentPoolProfiles {
+		if newCounts[old.Name] > old.Count {
+			scaleOps[old.Role] = scaleOp{up: true}
+			hasScaleOps = true
+			log.Debug("scale up")
+		} else if newCounts[old.Name] < old.Count {
+			scaleOps[old.Role] = scaleOp{down: true}
+			hasScaleOps = true
+			log.Debug("scale down")
+		}
+		oldCs.Properties.AgentPoolProfiles[i].Count = newCounts[old.Name]
```

Comment:
why are we mutating `oldCs` here?

Reply:
oh, for the deepEqual.  This seems bad, anything after this that uses the pointer isn't going to have accurate info

Reply:
Agreed, we are in need of deep copies to do these kind of operations safely. For now, this should be fine since once we figure out a difference in the counts, we don't care about old anymore.